include/pevent.hpp

```diff
@@ -20,6 +20,7 @@ typedef struct PEvent {
   int mapfd;               // FD for ring buffer, same as `fd` for perf events
   int attr_idx;            // matching perf_event_attr
   size_t ring_buffer_size; // size of the ring buffer
+  RingBufferType ring_buffer_type;
```

Comment:
I don't really like the duplication of `ring_buffer_type` and `ring_buffer_size` in `PEvent` and `RingBuffer`, but I am not sure how to improve this.

Reply:
The data we have is fine. Optionally, we could improve the grouping / naming.
We have the data to instrument the ring buffer and the instantiation of the ring buffer itself. We could make that clearer.